src/components/partnership/partnership.service.ts

```diff
@@ -494,6 +512,11 @@ export class PartnershipService {
     // mou start and end are now computed fields and do not get updated directly
     const object = await this.readOne(input.id, session);
 
+    this.verifyFundingType(
+      input.fundingType ?? object.fundingType.value,
```

Comment:
If a `fundingType` is not given, but the managing `type` is removed then this will error.
We should handle this edge case.

We should set the `fundingType` to `null` for them if they omit it and the new or existing `types` don't include managing.
If they pass in a `fundingType` and still want to remove the managing type we could either:
- Give an error (only managing can having funding type)
- Set it to null silently for them

I'm leaning towards the former so the API doesn't do something "unexpected". What do you think?

Reply:
That's great.
I will try to follow your steps.

Reply:
What is the best way to set the `fundingType` to `null` while updating properties.
```
Object.assign(input, {
 fundingType: null,
});
```
I tried above code but it's not working because it rejects if the value is null or same with origin value in the `database.service.ts`

Reply:
Well first you should not modify the `input`. If you need to change a property, create a new object.
```tsx
const changes = {
  ...input,
  fundingType: null,
};
```

Secondly, the database service is probably just ignoring nullable values. In most places we treat `null` & `undefined` the same. Which is why we do checks like `foo == null` instead of `===`. For these updates though, we want to treat `null` as an actual value and `undefined` as an omitted value.
My guess would be this line 
https://github.com/SeedCompany/cord-api-v3/blob/aff9e6a4554574d6c320c487e8cc105672da3397/src/core/database/database.service.ts#L450